merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/DirectiveTypeRegistry.java

```diff
@@ -0,0 +1,61 @@
+package gov.nasa.jpl.aerie.merlin.driver;
+
+import gov.nasa.jpl.aerie.merlin.protocol.driver.DirectiveTypeRegistrar;
+import gov.nasa.jpl.aerie.merlin.protocol.model.MissionModelFactory;
+import gov.nasa.jpl.aerie.merlin.protocol.model.TaskSpecType;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+
+public record DirectiveTypeRegistry<Registry, Model>(
+    Map<String, TaskSpecType<Model, ?, ?>> taskSpecTypes,
+    Registry registry)
+{
+  public static <Registry, Model>
+  DirectiveTypeRegistry<Registry, Model> extract(final MissionModelFactory<Registry, Model> factory) {
```

Comment:
FYI The commit called "Use buildRegistry instead of getTaskSpecTypes" does not compile because there's an extra generic in the `MissionModelFactory` parameter.

Reply:
Ah, you're right! (But one commit back, as discussed privately.)

Looks like the next commit compiles because the fix for this got added one commit too late. I've relocated that change to the correct commit.